admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/screen/dukan_requests/DukanRequestsInteractionListener.kt

```diff
@@ -0,0 +1,17 @@
+package net.thechance.mena.admin_panel.presentation.screen.dukan_requests
+
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+
+@OptIn(ExperimentalUuidApi::class)
+
+interface DukanRequestsInteractionListener {
+    fun onSortClicked(type: DukanRequestsScreenState.SortType)
+    fun onViewDetailsClicked(dukanId: Uuid)
+    fun onRetryClicked()
+    fun onPageChanged(page: Int)
+    fun onApproveClicked(dukanId: Uuid)
+    fun onRejectDialogClicked(dukanId: Uuid)
+    fun onRejectDialogCanceled()
+    fun onRejectConfirmed()
```

Comment:
```suggestion
    fun onApproveDukanClicked(dukanId: Uuid)
    fun onRejectDukanDialogClicked(dukanId: Uuid)
    fun onRejectDukanDialogCanceled()
    fun onRejectDukanConfirmed()
```

+ you can add another function for rejection message 
for example: 
```
fun onRejectionMessageChanged()
```